src/stream/tcp.rs

```diff
@@ -172,140 +192,130 @@ impl IpStackTcpStream {
 }
 
 impl AsyncRead for IpStackTcpStream {
-    fn poll_read(
-        mut self: std::pin::Pin<&mut Self>,
-        cx: &mut Context<'_>,
-        buf: &mut tokio::io::ReadBuf<'_>,
-    ) -> Poll<std::io::Result<()>> {
+    fn poll_read(mut self: std::pin::Pin<&mut Self>, cx: &mut Context<'_>, buf: &mut tokio::io::ReadBuf<'_>) -> Poll<std::io::Result<()>> {
         loop {
-            if self.tcb.retransmission.is_some() {
```

Comment:
Not exactly sure how retransmission is handled here. Is this one moved to the `PacketStatus::RetransmissionRequest => {...}` later in the same function?

Reply:
yes